### DISCUSSION TITLE
Setting a default Skin

### DISCUSSION THREAD

**Discussion by amitkot** [libktx/ktx#92] [Q&A] [answered]
Using 1.9.6-b6, I've been trying to set a default skin for my game, and I'm currently unclear as to what is the way to do it.

I followed the documentation and created a simple skin:
```kotlin
init {
  skin {
     textButton {
       font = BitmapFont()
       fontColor = Color.WHITE
  }
}
```

But then when I create a TextButton:
```kotlin
table {
  textButton("Tutorial")
}
```

I get an error that there is no `default` skin:
```
Exception in thread "LWJGL Application" com.badlogic.gdx.utils.GdxRuntimeException: No com.badlogic.gdx.scenes.scene2d.ui.TextButton$TextButtonStyle registered with name: default
```

When saving the skin as a variable and passing it to the button, it works fine.

**Comment by czyzby** [comment, marked as answer]
See the [Scene2D documentation](https://github.com/libktx/ktx/tree/master/scene2d). It's the first code snippet in the examples section. You have to set the `Scene2DSkin.defaultSkin` first in order to use methods without passing your skin explicitly.

```kotlin
init {
  Scene2DSkin.defaultSkin = skin {
    textButton {
      font = BitmapFont()
      fontColor = Color.WHITE
    }
  }
}
```

**Comment by amitkot** [comment]
I think I got the wrong impressing while reading the [Style](https://github.com/libktx/ktx/tree/master/style) docs that calling ```skin {}``` sets the default one, but I understand it better now.

Thanks!

**Comment by czyzby** [comment]
`skin` method is from a separate module and it's basically a `Skin` factory method, so you can initiate multiple instances in your application.

**Comment by amitkot** [comment]
Thanks!
I'm writing my first game using libGDX so there is quite a lot to learn.
So far KTX has been very helpful to me to thanks for all the hard work!

BTW, is there a good open source reference project you would suggest I study at to get a good idea of the the best practices?

**Comment by czyzby** [comment]
Best practices of LibGDX in general or just KTX? There are a few open-sourced KTX projects listed on [the official website](https://libktx.github.io/), but most of them are quickly hacked game jam projects that _just work_ (TM). I suggest looking into existing Java [LibGDX tutorials](https://github.com/jelhouss/libgdx_suit/) and gradually converting your code to Kotlin.
